airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_source_adapter.py

```diff
@@ -58,3 +68,63 @@ def _select_abstract_streams(self, config: Mapping[str, Any], configured_catalog
             if isinstance(stream_instance, AbstractStreamFacade):
                 abstract_streams.append(stream_instance.get_underlying_stream())
         return abstract_streams
+
+    def convert_to_concurrent_stream(self, logger: logging.Logger, stream: Stream, cursor: Optional[Cursor] = None) -> Stream:
+        """
+        Prepares a stream for concurrent processing by initializing or assigning a cursor,
+        managing the stream's state, and returning an updated Stream instance.
+        """
+        state: MutableMapping[str, Any] = {}
+
+        if cursor:
+            state = cursor.state
+
+            stream.cursor = cursor  # type: ignore[assignment]  # cursor is of type ConcurrentCursor, which inherits from Cursor
+            if hasattr(stream, "parent"):
+                stream.parent.cursor = cursor
+        else:
+            cursor = FinalStateCursor(
+                stream_name=stream.name,
+                stream_namespace=stream.namespace,
+                message_repository=self.message_repository,  # type: ignore[arg-type]  # _default_message_repository will be returned in the worst case
+            )
+        return StreamFacade.create_from_stream(stream, self, logger, state, cursor)
+
+    def initialize_cursor(
+        self,
+        stream: Stream,
+        state_manager: ConnectorStateManager,
+        converter: AbstractStreamStateConverter,
+        slice_boundary_fields: Optional[Tuple[str, str]],
+        start: Optional[CursorValueType],
+        end_provider: Callable[[], CursorValueType],
+        lookback_window: Optional[GapType] = None,
+        slice_range: Optional[GapType] = None,
+    ) -> Optional[ConcurrentCursor]:
+        lookback_window = lookback_window or timedelta(seconds=DEFAULT_LOOKBACK_SECONDS)
+
+        cursor_field_name = stream.cursor_field
+
+        if cursor_field_name:
+            if not isinstance(cursor_field_name, str):
+                raise ValueError(f"Cursor field type must be a string, but received {type(cursor_field_name).__name__}.")
+
+            stream_state = state_manager.get_stream_state(stream.name, stream.namespace)
+            cursor_field = CursorField(cursor_field_name)
```

Comment:
tiny tiny code style nit: Can we move these into the constructor directly? i don't think we need a var just to pass it in

Reply:
updated